Ubuntu Focal: Failed to access volume '/dev/nbd0p2': No such file or directory (draksetup postinstall)

BUG:
```
ntfs-3g: Failed to access volume '/dev/nbd0p2': No such file or directory
```

```
root@zen:~# draksetup postinstall
[2020-10-22 23:43:52,230][INFO] Determined PDB GUID: 81bc5c377c525081645f9958f20                                                                                                             9c5271
[2020-10-22 23:43:52,230][INFO] Determined kernel filename: ntkrnlmp.pdb
[2020-10-22 23:43:52,230][INFO] Fetching PDB file...
[2020-10-22 23:43:52,238][DEBUG] Starting new HTTPS connection (1): msdl.microso                                                                                                             ft.com:443
[2020-10-22 23:43:52,681][DEBUG] https://msdl.microsoft.com:443 "GET /download/s                                                                                                             ymbols/ntkrnlmp.pdb/81bc5c377c525081645f9958f209c5271/ntkrnlmp.pdb HTTP/1.1" 302                                                                                                              0
[2020-10-22 23:43:52,683][DEBUG] Starting new HTTPS connection (1): vsblobprodsc                                                                                                             ussu5shard7.blob.core.windows.net:443
[2020-10-22 23:43:53,793][DEBUG] https://vsblobprodscussu5shard7.blob.core.windo                                                                                                             ws.net:443 "GET /b-4712e0edc5a240eabf23330d7df68e77/F9658F231805BF03EE49D7392358                                                                                                             BEFE8ACBDF2D5143E81D4DE72FC3557F20D700.blob?sv=2019-02-02&sr=b&si=1&sig=J3m6snOd                                                                                                             bZxjF07bPkmN9MISO3xttsjhYe8NDxcDpMk%3D&spr=https&se=2020-10-23T23%3A58%3A25Z&rsc                                                                                                             l=x-e2eid-3868eb31-ec674d69-a3bb7424-e05b8a28-session-5d9249c1-2bab4019-8210b1ae                                                                                                             -947d4653 HTTP/1.1" 200 8596480
100%|█████████████████████████████████████| 8.60M/8.60M [00:07<00:00, 1.11MiB/s]
[2020-10-22 23:44:01,556][INFO] Generating profile out of PDB file...
[2020-10-22 23:44:10,726][INFO] Saving profile...
[2020-10-22 23:44:11,478][INFO] Saving runtime profile...
[2020-10-22 23:44:11,479][INFO] Saving VM snapshot...
Saving to /var/lib/drakrun/volumes/snapshot.sav new xl format (info 0x3/0x0/2350                                                                                                             )
xc: info: Saving domain 3, type x86 HVM
xc: Frames: 1044480/1044480  100%
xc: End of stream: 0/0    0%
[2020-10-22 23:44:19,638][INFO] Snapshot was saved succesfully.
ntfs-3g: Failed to access volume '/dev/nbd0p2': No such file or directory

ntfs-3g 2017.3.23AR.3 integrated FUSE 28 - Third Generation NTFS Driver
                Configuration type 7, XATTRS are on, POSIX ACLS are on

Copyright (C) 2005-2007 Yura Pakhuchiy
Copyright (C) 2006-2009 Szabolcs Szakacsits
Copyright (C) 2007-2017 Jean-Pierre Andre
Copyright (C) 2009 Erik Larsson

Usage:    ntfs-3g [-o option[,...]] <device|image_file> <mount_point>

Options:  ro (read-only mount), windows_names, uid=, gid=,
          umask=, fmask=, dmask=, streams_interface=.
          Please see the details in the manual (type: man ntfs-3g).

Example: ntfs-3g /dev/sda1 /mnt/windows

News, support and information:  http://tuxera.com
[2020-10-22 23:44:20,476][WARNING] Generating usermode profiles failed
[2020-10-22 23:44:20,477][ERROR] Failed to mount /dev/nbd0p2 as NTFS.
Traceback (most recent call last):
  File "/opt/venvs/drakrun/lib/python3.8/site-packages/drakrun/draksetup.py", li                                                                                                             ne 217, in create_rekall_profiles
    subprocess.check_output(f"mount -t ntfs -o ro {blk_quoted} {mnt_path_quoted}                                                                                                             ", shell=True)
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'mount -t ntfs -o ro /dev/nbd0p2 /tmp/tmp                                                                                                             zjf0qjbz' returned non-zero exit status 11.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venvs/drakrun/lib/python3.8/site-packages/drakrun/draksetup.py", li                                                                                                             ne 324, in postinstall
    create_rekall_profiles(install_info)
  File "/opt/venvs/drakrun/lib/python3.8/site-packages/drakrun/draksetup.py", li                                                                                                             ne 219, in create_rekall_profiles
    raise RuntimeError(f"Failed to mount {block_device} as NTFS.")
RuntimeError: Failed to mount /dev/nbd0p2 as NTFS.
[2020-10-22 23:44:20,610][INFO] Generated VM configuration for vm-0
[2020-10-22 23:44:20,711][INFO] Generated VM configuration for vm-1
[2020-10-22 23:44:20,716][DEBUG] Starting new HTTPS connection (1): drakvuf.iced                                                                                                             ev.pl:443
[2020-10-22 23:44:21,009][DEBUG] https://drakvuf.icedev.pl:443 "POST /usage/drak                                                                                                             setup HTTP/1.1" 200 16
[2020-10-22 23:44:22,838][INFO] Enabling and starting drakrun@1...
[2020-10-22 23:44:24,059][INFO] All right, drakrun setup is done.
```

But afterwards `/dev/nbd0p2` appeared and I'm able to mount it.

/cc @BonusPlay: Is there some race condition that we bind `/dev/nbd0` but it actually happens asynchronously or we just simply don't wait for partitions to be detected?

--- a/drakrun/drakrun/storage.py
+++ b/drakrun/drakrun/storage.py
@@ -253,7 +253,17 @@ class Qcow2StorageBackend(StorageBackendBase):
             raise RuntimeError("Failed to connect QCOW2 file to /dev/nbd0")
 
         # we mount 2nd partition, as 1st partition is windows boot related and 2nd partition is C:\\
-        yield "/dev/nbd0p2"
+        dev = "/dev/nbd0p2"
+
+        # wait for the device to appear
+        for _ in range(60):
+            if os.path.exists(dev):
+                break
+            time.sleep(1.0)
+        else:
+            raise RuntimeError(f"NBD volume not available at {dev}")
+
+        yield dev
 
         subprocess.check_output("qemu-nbd --disconnect /dev/nbd0", shell=True)
 

